### PR TITLE
Fix update background_image of a Collection in GraphQL API

### LINKED ISSUE #3269: GraphQL cannot update background_image of a collection
### What I'm trying to achieve
Upload a new `background_image` to a collection in dashboard 2.0.

### Steps to reproduce the problem
1. Edit a collection in dashboard 2.0.;
2. Upload a new background image;
3. You should get: `ERROR graphql.execution.utils graphql.error.located_error.GraphQLLocatedError: 'name'`

### LINKED ISSUE #3384: Can't delete background_image of a collection in GrapQL
### What I'm trying to achieve
I want to have a functionality which allows staff to delete background_image of a collection.

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #3269, resolve #3384

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.

### CODE REVIEW

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def user_is_allowed(cls, user, input):
     @classmethod
     def clean_input(cls, info, instance, input, errors):
         cleaned_input = super().clean_input(info, instance, input, errors)
-        if 'slug' not in cleaned_input:
+        if 'slug' not in cleaned_input and 'name' in cleaned_input:
```
Comment (Pacu2) [inline saleor/graphql/product/mutations/products.py]: What if there's no name provided? What will be the slug?

**Comment by fowczarek** [comment]
@dominik-zeglen 
 The Deletion button does not trigger any action in Dashboard 2.0.
Can you fix this?

**Comment by NyanKiyoshi** [comment]
@fowczarek you may want to put a keyword before the second issue reference (#3384), GitHub didn't flag it as to be closed once merged. :+1:

**Comment by dominik-zeglen** [comment]
Would you mind updating `Upload` graphql type's description?

**Pull request merged**